packages/react-reconciler/src/ReactFiberRoot.new.js

```diff
@@ -46,12 +46,13 @@ function FiberRootNode(containerInfo, tag, hydrate) {
   this.pingedLanes = NoLanes;
   this.expiredLanes = NoLanes;
   this.mutableReadLanes = NoLanes;
-
   this.finishedLanes = NoLanes;
 
   this.entangledLanes = NoLanes;
   this.entanglements = createLaneMap(NoLanes);
 
+  this.mutableSourceEagerHydrationData = [];
```

Comment:
This array is a bit unnecessary if we're not going to hydrate. In the case that you use many roots, it's probably because they're not hydrating. Maybe we can make this nullable?

Additionally, if this is a non-hydration compatible renderer, we probably shouldn't add this field at all.

Can you gate all of this code behind the `supportsHydration` HostConfig? That way we can produce an optional DOM build that strips out hydration easily.

Reply:
Sure.